gui/src/store/contracts.ts

```diff
@@ -0,0 +1,73 @@
+import { invoke } from "@tauri-apps/api/tauri";
+import { Address } from "viem";
+import { StateCreator, create } from "zustand";
+
+import ABI from "../erc20.json";
+import { ABIItem, IContract } from "../types";
+
+const API_KEY = import.meta.env.VITE_ETHERSCAN_API_KEY;
+
+interface Store {
+  data: Record<number, Array<IContract>>;
+
+  init: () => void;
+  addAddress: (chainId: number, address: Address) => void;
+}
+
+const store: StateCreator<Store> = (set, get) => ({
+  data: {},
+
+  init: async () => {
+    const contracts = await invoke<IContract[]>("db_get_contracts", {
+      chainId: 31337,
+    });
+    set({
+      data: {
+        31337: contracts.map(({ address }) => ({
+          address,
+          abi: ABI as ABIItem[],
+          name: "ERC20",
```

Comment:
note to self: these two lines need to change. get the actual ABI from the backend instead of the hardcoded erc20